app/javascript/packs/OAS3Autocomplete.js

```diff
@@ -53,25 +53,36 @@ const updateResponseBody = (response: SwaggerResponse, accountData: AccountData)
   }
 )
 
+const injectServerToResponseBody = (responseBody: ResponseBody, serviceEndpoint: string): ResponseBody => {
+  const originalServers = responseBody.servers ? responseBody.servers : []
+  const servers = serviceEndpoint ? [{url: serviceEndpoint}] : originalServers
+
+  return {
+    ...responseBody,
+    servers
+  }
+}
+
 // response.body.method is not present when fetching the spec,
 // is present when doing a request to one of the paths
 const isSpecFetched = (response: SwaggerResponse): boolean => !!response.body.method
 
-export const autocompleteOAS3 = async (response: SwaggerResponse, accountDataUrl: string): SwaggerResponse => (
-  isSpecFetched(response)
-    ? response
-    : fetchData(accountDataUrl)
-      .then(data => {
-        const accountData = data.results
-        if (!accountData) {
-          return response
-        }
-        const body = updateResponseBody(response, accountData)
-        return {
-          ...response,
-          body,
-          data: JSON.stringify(body),
-          text: JSON.stringify(body)
-        }
-      })
-)
+export const autocompleteOAS3 = async (response: SwaggerResponse, accountDataUrl: string, serviceEndpoint: string): SwaggerResponse => {
+  if (isSpecFetched(response)) {
+    return response
+  }
+
+  let body = injectServerToResponseBody(response.body, serviceEndpoint)
+
+  const accountData = await fetchData(accountDataUrl).then(data => data)
+  if (accountData.results) {
+    body = injectAutocompleteToResponseBody(body, accountData.results)
+  }
```

Comment:
```suggestion
  // The params of `injectServerToResponseBody` and `injectAutocompleteToResponseBody` should be curried
  
  const injectServer = injectServerToResponseBody(serviceEndpoint)

  const body = await fetchData(accountDataUrl).then(data => (
   data.results 
     ? [injectServer, injectAutocompleteToResponseBody(data.results)].reduce((body, fn) => fn(body), response.body) 
     : injectServer(response.body)
  ))
```

Reply:
There is something wrong with this, it breaks the flow
I think it's because `injectServerToResponseBody` is getting the wrong number of args, I will take a look later and we can make an inprovement later, by now we can keep what is working